### ISSUE TITLE
Compatibility with View Component

### ISSUE THREAD

**Issue by joemasilotti** [glebm/i18n-tasks#452]
I can't get i18n-tasks to recognize strings in my View Components when using the shorthand dot syntax.

```ruby
# app/components/my_component.rb
class MyComponent < ViewComponent::Base
  def foo
    t(".key")
  end
end

# app/components/my_component.html.erb
<%= foo %>

# config/locales/en.yml
en:
  my_component:
    key: English translation
```

```bash
$ bundle exec i18n-tasks health en

Unused keys (1) | i18n-tasks v1.0.9
+--------+------------------+---------------------+
| Locale | Key              | Value               |
+--------+------------------+---------------------+
|   en   | my_component.key | English translation |
+--------+------------------+---------------------+
```

I have the following relevant configuration. 

```ruby
# config/i18n-tasks.yml
relative_exclude_method_name_paths:
  - app/components
```

---

Removing the `relative_exclude_method_name_paths` configuration the health check output _adds_ the following ("Unused keys" remains).

```bash
Missing translations (1) | i18n-tasks v1.0.9
+--------+----------------------+----------------------------------+
| Locale | Key                  | Value in other locales or source |
+--------+----------------------+----------------------------------+
|   en   | my_component.foo.key | app/components/my_component.rb:3 |
+--------+----------------------+----------------------------------+
```

**Comment by davidwessman** [comment]
@glebm I cannot find any code for understanding what a ViewComponent.
Is there any default behaviour that should catch these paths?

**Comment by davidwessman** [comment]
> @davidwessman [ViewComponent](https://viewcomponent.org/) docs can be found here. It's a framework from GitHub that is gaining traction across the Rails community.

Ah, yes - thank you! But I was thinking about in the I18n-tasks codebase.

**Comment by joemasilotti** [comment]
I don't think there's anything in the codebase _preventing_ compatibility. My guess is that there's something not working with inferring the correct keys when matching strings.

**Comment by joemasilotti** [comment]
That's what I have already. Unless order of `relative_roots` matters?

https://github.com/joemasilotti/railsdevs.com/blob/main/config/i18n-tasks.yml#L60-L74

**Comment by davidwessman** [comment]
> That's what I have already. Unless order of `relative_roots` matters?
> 
> https://github.com/joemasilotti/railsdevs.com/blob/main/config/i18n-tasks.yml#L60-L74

It was not included in the first comment on the issue, so I wanted to make sure.
The order should not matter as far as I understand the code.

Will see if I can reproduce it with a test case when I get time 🙂

**Comment by joemasilotti** [comment]
Sounds good, thanks! The repo I linked to is open source if it helps reproduce.

**Comment by davidwessman** [comment]
I did some more research on this regarding what to expect from ViewComponents and I am not 100% what we want I18n-tasks to do here. So the current state is using global vs local translations, not using namespaces as we might otherwise do:
https://github.com/github/view_component/blob/main/docs/guide/translations.md

In this example:
```ruby
# app/components/my_component.rb
class MyComponent < ViewComponent::Base
  def foo
    t(".key") 
  end
end
```
it seems like the docs expect it to translate to:

```yaml
# app/components/my_component.en.yml
en:
  key: "What"
```

So if i18n-tasks translates that to `en.my_component.key`, wouldn't that generate false results as well?
What do you think @glebm? Should we implement something explicit for `ViewComponent` that matches their documented behaviour? They do not document anything about using `t` inside the component itself, only the view.

I will see if I can get it to work when adding it to `relative_roots`, but not make it `ViewComponent`-specific.

**Comment by davidwessman** [comment]
I found a bug in the use of this option, @joemasilotti does it work with:

```diff
 relative_roots:
   - app/controllers
   - app/helpers
   - app/mailers
   - app/presenters
   - app/views
   - app/components
   
-relative_exclude_method_name_paths:
+exclude_method_name_paths:
  - app/components
```

It should probably stay as the documented `relative_exclude_method_name_paths` when there is a new version out.

**Comment by davidwessman** [comment]
> I did some more research on this regarding what to expect from ViewComponents and I am not 100% what we want I18n-tasks to do here. So the current state is using global vs local translations, not using namespaces as we might otherwise do: https://github.com/github/view_component/blob/main/docs/guide/translations.md
> 
> In this example:
> 
> ```ruby
> # app/components/my_component.rb
> class MyComponent < ViewComponent::Base
>   def foo
>     t(".key") 
>   end
> end
> ```
> 
> it seems like the docs expect it to translate to:
> 
> ```yaml
> # app/components/my_component.en.yml
> en:
>   key: "What"
> ```
> 
> So if i18n-tasks translates that to `en.my_component.key`, wouldn't that generate false results as well? What do you think @glebm? Should we implement something explicit for `ViewComponent` that matches their documented behaviour? They do not document anything about using `t` inside the component itself, only the view.
> 
> I will see if I can get it to work when adding it to `relative_roots`, but not make it `ViewComponent`-specific.

I tried setting up a project with ViewComponent, and the raised errors for a missing translation was:

- `t(".relative_key")` in `app/components/my_component.rb` =>  `en.my_component.relative_key`
- `t(".relative_key")` in `app/components/my_component.html.erb` =>  `en.my_component.relative_key`

So I think it makes sense to rely on adding `app/components` to `relative_roots` and `relative_exclude_method_name_paths` when using `ViewComponent`.

**Comment by glebm** [comment]
Released v1.0.10

**Comment by joemasilotti** [comment]
Works great for me, thanks for the quick turnaround on this!
